File: src/main/java/kitchenpos/menus/tobe/domain/JpaMenuRepository.java
package kitchenpos.menus.tobe.domain;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;
import java.util.UUID;

public interface JpaMenuRepository extends MenuRepository, JpaRepository<kitchenpos.menus.tobe.domain.Menu, UUID> {
    @Query("select m from Menu m join m.menuProducts mp where mp.product.id = :productId")
    @Override
    List<kitchenpos.menus.tobe.domain.Menu> findAllByProductId(@Param("productId") UUID productId);
}

File: src/main/java/kitchenpos/menus/tobe/domain/JpaMenuRepository.kt
package kitchenpos.menus.tobe.domain

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface JpaMenuRepository : JpaRepository<Menu, UUID>, MenuRepository {
    @Query("select m from Menu m join m.menuProducts mp where mp.productId = :productId")
    override fun findAllByProductId(productId: UUID): List<Menu>
}
